src/racer/scopes.rs

```diff
@@ -228,6 +228,7 @@ pub fn get_start_of_search_expr(src: &str, point: usize) -> usize {
             (b'.', State::MustEndsWithDot(_)) =>  State::None,
             (b':', State::MustEndsWithDot(index)) =>  State::StartsWithCol(index),
             (b':', State::StartsWithCol(_)) =>  State::None,
+            (b'?', State::StartsWithDot) => State::None,
```

Comment:
I think the current impl is fine. `?` isn't really a search expr char, there's no completions to make until you attempt field access.